app/workers/share_worker.rb

```diff
@@ -23,11 +22,25 @@ def perform(user_id, answer_id, service)
     logger.info "Tried to post answer ##{answer_id} from user ##{user_id} to Twitter but the account is suspended."
   rescue Twitter::Error::Unauthorized
     # User's Twitter token has expired or been revoked
-    # TODO: Notify user if this happens (https://github.com/Retrospring/retrospring/issues/123)
     logger.info "Tried to post answer ##{answer_id} from user ##{user_id} to Twitter but the token has exired or been revoked."
+    revoke_and_notify(user_id, service)
   rescue => e
     logger.info "failed to post answer #{answer_id} to #{service} for user #{user_id}: #{e.message}"
     Sentry.capture_exception(e)
     raise
   end
+
+  def revoke_and_notify(user_id, service)
+    user_service = find_service(user_id, service)
+    user_service.destroy
```

Comment:
...which can then be used here to destroy the service

```suggestion
    @user_service.destroy
```

-> 1 query less!